pkg/client/observatorium/api.go

```diff
@@ -203,17 +209,33 @@ func (obs *ServiceObservatorium) GetMetrics(metrics *KafkaMetrics, namespace str
 		},
 	}
 
-	var failedMetrics []string
+	// build query per label to reduce query count
 	queries := obs.buildQueries(fetchers, rq)
+
+	// distribute metrics queries to observatorium
+	resultChan := make(chan metricQueryResult)
 	for _, query := range queries {
-		result := obs.fetchMetricsResult(query, rq)
-		if result.Err != nil {
-			failedMetrics = append(failedMetrics, fmt.Sprintf("%q:%v", query, result.Err))
-			glog.Errorf("error running query %q: %v", query, result.Err)
+		go func(query string) {
+			metricResult := obs.fetchMetricsResult(query, rq)
+			resultChan <- metricQueryResult{
+				metricResult: metricResult,
+				metricQuery:  query,
+			}
+		}(query)
+	}
+
+	// process the metrics result
+	var failedMetrics []string
+	for i := 1; i <= len(queries); i++ {
+		metricQueryResult := <-resultChan
```

Comment:
I understand this will block until it receives a result right? We can guarantee we will wait until all metrics have been queried due to the length of the queries slice

Reply:
Also the order we will receive the results is not guaranteed to be the order of the original metrics. Is that an issue?

Reply:
Yes, that is correct. No metrics can be skipped, even in the case of an error, a result will be returned from the query. So there should always be as many items on the channel as there are queries.

Reply:
> Yes, this was the intention. Do you think we should use the WaitGroup as well?

I don't think that's needed as the channel blocks and the length should guarantee all of them will be processed

Reply:
Thanks @pb82 & @miguelsorianod

Reply:
The order won't be an issue for the e2e tests and if it were, I'd consider that a problem with the test that needs to be addressed.

Reply:
> The order won't be an issue for the e2e tests and if it were, I'd consider that a problem with the test that needs to be addressed.

+1